app/Models/User.php

```diff
@@ -609,4 +607,52 @@ public function scopeTracked(Builder $query): Builder
     {
         return $query->where('Untracked', false); // TODO: use unranked_at=NULL?
     }
+
+    /**
+     * @param Builder<User> $query
+     * @return Builder<User>
+     */
+    public function scopeSearch(Builder $query, string $keyword): Builder
+    {
+        // Start with base conditions that apply to all searches.
+        $baseQuery = $query
+            ->whereNotNull('email_verified_at')
+            ->whereNull('Deleted')
+            ->whereNull('banned_at');
+
+        // Return early if keyword is empty to avoid unnecessary processing.
+        if (empty($keyword)) {
+            return $baseQuery;
+        }
+
+        // Common ranking calculation for both queries.
+        $activityRank = "
+            CASE
+                WHEN achievements_unlocked > 0 THEN 40
+                WHEN LastLogin > DATE_SUB(NOW(), INTERVAL 30 DAY) THEN 20
+                WHEN LastLogin > DATE_SUB(NOW(), INTERVAL 90 DAY) THEN 10
+                ELSE 0
+            END";
+
+        return $baseQuery
+            ->select('*')
+            ->selectRaw("
+                CASE 
+                    WHEN User = ? THEN 100
+                    WHEN display_name = ? THEN 90
+                    WHEN User LIKE ? THEN 80
+                    WHEN display_name LIKE ? THEN 70
+                    ELSE 0
+                END + {$activityRank} as search_rank
+            ", [$keyword, $keyword, "{$keyword}%", "{$keyword}%"])
+            ->where(function ($q) use ($keyword) {
+                $q->where('User', '=', $keyword)
+                  ->orWhere('display_name', '=', $keyword)
+                  ->orWhere('User', 'LIKE', "{$keyword}%")
+                  ->orWhere('display_name', 'LIKE', "{$keyword}%");
+            })
+            ->orderBy('search_rank', 'desc')
+            ->orderBy('achievements_unlocked', 'desc')
+            ->limit(10);
```

Comment:
The documentation for a FULLTEXT index says you have to use match/against. Is this smart enough to do that?

Reply:
Nope! Thanks for catching this, I messed up before I pushed and forgot to revert back to the correct query. Fixed in latest.